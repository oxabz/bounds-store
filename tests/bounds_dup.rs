#[allow(dead_code)]
use bounds_store::{bounds, bound_alias};

trait Float {}

type Point<F: Float> = (F, F);

trait Polygon<'a, F: Float>: 'a + IntoIterator<Item = &'a Point<F>> where F: 'a {}

bounds! {
    Polygon => <'a, F: 'a + Float, P: Polygon<'a, F>>
}

bounds! {
    Polygon => <'a, F: 'a + Float, P: Polygon<'a, F>>
}

#[bound_alias(Polygon)]
fn area(_polygon: P) -> F {
    unimplemented!()
}

fn main() {}