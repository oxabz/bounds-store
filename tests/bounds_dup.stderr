error: proc macro panicked
  --> tests/bounds_dup.rs:14:1
   |
14 | / bounds! {
15 | |     Polygon => <'a, F: 'a + Float, P: Polygon<'a, F>>
16 | | }
   | |_^
   |
   = help: message: bounds! can only be called once
